invenio_app_ils/circulation/receivers.py

```diff
@@ -12,18 +12,23 @@
 from invenio_circulation.signals import loan_state_changed
 
 from ..proxies import current_app_ils_extension
-from ..records.api import Item
+from ..records.api import Item, Document
 
 
 def register_circulation_signals(app):
     """Register Circulation signal."""
-    loan_state_changed.connect(index_item_after_loan_change,
+    loan_state_changed.connect(index_record_after_loan_change,
                                sender=app, weak=False)
 
 
-def index_item_after_loan_change(_, loan):
+def index_record_after_loan_change(_, loan):
```

Comment:
This is more food for thought than a request. We will possibly have to index more things than item and document and we will have more cases that we need to trigger manual indexing. My proposal would be each model to have an individual function 

`def index_item(data):...`
`def index_doc(data):...`

and then compose the indexers according to the case:
```
def index_record_after_loan_change(_, loan):
    index_item(loan)
    index_doc(loan)
    ...
```

Reply:
Yes but in this particular case here is just the indexing after the loan signal fired. For the other cases I will create a specific indexer per record type (e.g item, document) which will be responsible for the indexing and all the side effects.